src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -378,32 +386,89 @@ public void onTraversalStart() {
         }
 
         // Check for reference version (in)compatibility:
-        if ( allowHg19ContigNamesWithB37 &&
-             referenceVersion.equals(FuncotatorArgumentDefinitions.HG19_REFERENCE_VERSION_STRING) ) {
+        if ( forceB37ToHg19ContigNameConversion || referenceVersion.equals(FuncotatorArgumentDefinitions.HG19_REFERENCE_VERSION_STRING) ) {
```

Comment:
I think that you need to check in `onTraversalStart()` that the reference passed in via `-R` and the sequence dictionary for the set of variants you're annotating (`-V`) are both B37 or both HG19, if either of them is B37/HG19. There is a sequence dictionary compatibility check in the engine, but it's not necessarily sufficient to detect the case of an HG19 fasta + a B37 VCF (for example), since it only checks that there is at least one shared contig between dictionaries.

It would also be good (while you're on this topic) to explore whether the current check in `GATKTool.validateSequenceDictionaries()` would sometimes conclude that a B37 and HG19 dictionary were compatible. If it would, you should file a ticket against it.

Reply:
I'll add this check in the form of a call to `SequenceDictionaryUtils.validateDictionaries` that will require the reference sequence dictionary to be a superset of the variant sequence dictionary.

I will need to fix many tests that have broken because of this (and change some file names to be more descriptive).


Also:
I've verified that giving `b37` variants with the UCSC hg19 reference causes `validateSequenceDictionaries` to get upset (this is correct).

It does not get upset when I give `b37` variants and the GRCh37 reference (it should).